packages/remix/src/plugins/plugin.ts

```diff
@@ -0,0 +1,212 @@
+      getLockFileName(detectPackageManager(context.workspaceRoot)),
+    ]);
+    const targets = targetsCache[hash]
+      ? targetsCache[hash]
+      : await buildRemixTargets(configFilePath, projectRoot, options, context);
+
+    calculatedTargets[hash] = targets;
+
+    return {
+      projects: {
+        [projectRoot]: {
+          root: projectRoot,
+          targets,
+        },
+      },
+    };
+  },
+];
+
+async function buildRemixTargets(
+  configFilePath: string,
+  projectRoot: string,
+  options: RemixPluginOptions,
+  context: CreateNodesContext
+) {
+  const namedInputs = getNamedInputs(projectRoot, context);
+  const serverBuildPath = await getServerBuildPath(
+    configFilePath,
+    context.workspaceRoot
+  );
+
+  const targets: Record<string, TargetConfiguration> = {};
+  targets[options.buildTargetName] = buildTarget(
+    options.buildTargetName,
+    namedInputs
+  );
+  targets[options.serveTargetName] = serveTarget(serverBuildPath);
+  targets[options.startTargetName] = startTarget(
+    projectRoot,
+    serverBuildPath,
+    options.buildTargetName
+  );
+  targets[options.typecheckTargetName] = typecheckTarget(
+    projectRoot,
+    namedInputs
+  );
+
+  return targets;
+}
+
+function buildTarget(
+  buildTargetName: string,
+  namedInputs: { [inputName: string]: any[] }
+): TargetConfiguration {
+  return {
+    cache: true,
+    dependsOn: [`^${buildTargetName}`],
+    inputs: [
+      ...('production' in namedInputs
+        ? ['production', '^production']
+        : ['default', '^default']),
+    ],
+    outputs: ['{options.outputPath}'],
+    executor: '@nx/remix:build',
+    options: {
+      outputPath: '{workspaceRoot}/dist',
+    },
+  };
+}
+
+function serveTarget(serverBuildPath: string): TargetConfiguration {
+  return {
+    executor: '@nx/remix:serve',
+    options: {
+      command: `npx remix-serve ${serverBuildPath}`,
+      port: 4200,
+    },
+  };
+}
+
+function startTarget(
+  projectRoot: string,
+  serverBuildPath: string,
+  buildTargetName: string
+): TargetConfiguration {
+  return {
+    dependsOn: [buildTargetName],
+    command: `npx remix-serve ${serverBuildPath}`,
+    options: {
+      cwd: projectRoot,
+    },
+  };
+}
+
+function typecheckTarget(
+  projectRoot: string,
+  namedInputs: { [inputName: string]: any[] }
+): TargetConfiguration {
+  return {
+    cache: true,
+    inputs: [
+      ...('production' in namedInputs
+        ? ['production', '^production']
+        : ['default', '^default']),
+    ],
+    command: 'tsc',
+    options: {
+      cwd: projectRoot,
+    },
+  };
+}
+
+async function getServerBuildPath(
+  configFilePath: string,
+  workspaceRoot: string
+): Promise<string> {
+  const configPath = join(workspaceRoot, configFilePath);
+  let appConfig: AppConfig = {};
+  try {
+    let appConfigModule: any;
+    try {
+      appConfigModule = require(configPath);
+    } catch {
+      appConfigModule = await Function(`return import("${configPath}")`)();
```

Comment:
We can always use `import()` and not need the try-catch here, since it works with both CJS and ESM.

Reply:
Let me double check it, but i'm pretty sure i was running into an issue with using import on the CJS config which led me to this.

Reply:
It's because of Jest.

It throws 
```
TypeError [ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING]: A dynamic import callback was not specified.
```
when just using import.

Reply:
I will swap it around however, because import will work and has worked in all my testing and the require should only be required for jest